src/core/ayayaa-server/ayayaa-logging/Interfaces/ILogger.cs

```diff
@@ -0,0 +1,17 @@
+﻿using ayayaa.logging.Enums;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ayayaa.logging.Interfaces
+{
+    public interface ILogger
+    {
+        Dictionary<IWriter, LogPriority> Writers { get; }
+
+        void AddWriter(IWriter writer, LogPriority priority);
+        void WriteToLog(string message, LogPriority priority, IWriter log);
+        Dictionary<IWriter, bool> WriteToLogs(string message, LogPriority priority);
```

Comment:
Returning a dictionary of success/failure is probably overkill here. A single boolean to indicate if the log was written by *any* writer would be ok.

Considering that most log output will never be error checked.